src/librustdoc/html/format.rs

```diff
@@ -761,6 +761,9 @@ fn fmt_type<'cx>(
             fmt::Display::fmt(&tybounds(bounds, lt, cx), f)
         }
         clean::Infer => write!(f, "_"),
+        clean::Primitive(clean::PrimitiveType::Never) => {
+            primitive_link(f, PrimitiveType::Never, "!", cx)
+        }
```

Comment:
Are there other places where there needs to be a special-case to replace `never` with `!`?

Reply:
Normally they're all here.